### PR TITLE
[tune] tbx logger

### PR DESCRIPTION
**Pull request by richardliaw**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/tune/logger.py [hunk 1]
```diff
@@ -316,6 +316,58 @@ def close(self):
         self._file.close()
 
 
+class TBXLogger(Logger):
+    """TensorBoardX Logger, because TF breaks everything.
```
Comment (ericl) [inline python/ray/tune/logger.py]: lol. Please fix before merging.

File: python/ray/tune/logger.py [hunk 2]
```diff
@@ -316,6 +316,58 @@ def close(self):
...
+
+    Automatically flattens nested dicts to show on TensorBoard:
+
+        {"a": {"b": 1, "c": 2}} -> {"a/b": 1, "a/c": 2}
+    """
+
+    def _init(self):
+        from tensorboardX import SummaryWriter
```
Comment (ericl) [inline python/ray/tune/logger.py]: Should we make this a mandatory dep?

**Comment by AmeerHajAli** [comment]
Thanks for the fix! It resolved my issue with tensorboard (TF2).

File: python/ray/tune/logger.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/tune/logger.py]: No I think not. It has issues of its own...

File: doc/source/tune-usage.rst [hunk 3]
```diff
@@ -598,13 +598,29 @@ You can pass in your own logging mechanisms to output logs in custom formats as
...
     )
 
 These loggers will be called along with the default Tune loggers. All loggers must inherit the `Logger interface <tune-package-ref.html#ray.tune.logger.Logger>`__. Tune enables default loggers for Tensorboard, CSV, and JSON formats. You can also check out `logger.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/logger.py>`__ for implementation details. An example can be found in `logging_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/logging_example.py>`__.
 
+.. warning:: If you run into issues for TensorBoard logging, consider using the TensorBoardX Logger (``from ray.tune.logger import TBXLogger``)
+
+TBXLogger (TensorboardX)
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tune provides a logger using `TensorBoardX <https://github.com/lanpa/tensorboardX>`_. You can install tensorboardX via ``pip install tensorboardX``. This logger automatically outputs loggers similar to the default TensorFlow logging format but is nice if you are undergoing a TF1 to TF2 transition. By default, it will log any scalar value provided via the result dictionary along with HParams information.
```
Comment (ericl) [inline doc/source/tune-usage.rst]: Why not make this the default TF logger, remove the others, and make it a required package for Tune and RLlib?

Comment (richardliaw) [reply doc/source/tune-usage.rst]: I mean I think that's a good point; I'm just worried about the issues it has on its own (i.e., https://github.com/lanpa/tensorboardX/issues/532, etc).

**Review by ericl — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by richardliaw** [comment]
jenkins test tune

**Pull request merged**